File: MiniTicketSystem/Entities/Ticket.cs
using MiniTicketSystem.Entities;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MiniTicketSystem.Entities;

public class Ticket
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    public required string Title { get; set; }

    [Required]
    public required string Description { get; set; }

    public required Guid StatusId { get; set; }

    public Status Status { get; set; } = null!;
}

File: mini-ticket-system/src/app/models/ticket.ts
export interface Ticket {
  id: string;
  title: string;
  description: string;
  statusId: string;
  statusName: string;
}

export interface CreateTicketDto {
  title: string;
  description: string;
  statusId: string;
}

export interface UpdateTicketStatusDto {
  id: string;
  title: string;
  description: string;
  statusId: string;
}
